File: Chapter 03/Python/3.4 OrderProcessor Refactor/After/order_service.py
class OrderService:
    def __init__(self, validator, payment_service, inventory_manager, notification_service):
        self.validator = validator
        self.payment_service = payment_service
        self.inventory_manager = inventory_manager
        self.notification_service = notification_service

    def process_order(self, order):
        self.validator.validate(order)

        if self.payment_service.process_payment(order):
            self.inventory_manager.update_inventory(order)
            self.notification_service.send_confirmation_email(order)
            return "Order processed successfully."
        else:
            return "Payment failed."

File: Chapter 03/JS/3.4 OrderProcessor Refactor/After/orderService.js
class OrderService {
    constructor(validator, paymentService, inventoryManager, notificationService) {
        this.validator = validator;
        this.paymentService = paymentService;
        this.inventoryManager = inventoryManager;
        this.notificationService = notificationService;
    }

    processOrder(order) {
        this.validator.validate(order);

        if (this.paymentService.processPayment(order)) {
            this.inventoryManager.updateInventory(order);
            this.notificationService.sendConfirmationEmail(order);
            return "Order processed successfully.";
        } else {
            return "Payment failed.";
        }
    }
}
module.exports = OrderService;